topo_processor/util/aws_credentials.py

```diff
@@ -3,11 +3,32 @@
 import boto3
 
 credentials = None
+role_dict = {}
 
 
-def get_credentials():
+def get_credentials(bucket_name: str):
     global credentials
-    if not credentials:
-        session = boto3.Session(profile_name=os.getenv("AWS_PROFILE"))
-        credentials = session.get_credentials()
+    if credentials:
```

Comment:
this check cannot be done before the `if bucket_name in role_dict` check.

as it will ignore the bucket specific checks.